04-onion-routing.md

```diff
@@ -601,6 +606,20 @@ If the `cltv-expiry` is too low, the final node MUST fail the HTLC:
 
 1. type: 17 (`final_expiry_too_soon`)
 
+If the `outgoing_cltv_value` does not match the `ctlv-expiry` of the
+HTLC at the final hop:
+
+1. type: 18 (`final_incorrect_cltv_expiry`)
+2. data:
+   * [4:cltv-expiry]
+
+If the `amt_to_forward` does not match the `incoming_htlc_amt` of
```

Comment:
Oh, you're right, I got it mixed up, we're only saying how much to forward, not the fee. So fee shaving in the hopes of being the second to last hop is indeed a problem, since it would result in failed payments if we're not the second to last hop. Happy to merge your version as is :wink: